Logic/Prover.hs

```diff
@@ -88,6 +89,13 @@ joinSensAux s1 s2 = let
 joinSens :: (Ord a, Eq b) => ThSens a b -> ThSens a b -> ThSens a b
 joinSens s1 = fst . joinSensAux s1
 
+intersectSens :: (Ord a, Eq b) => ThSens a b -> ThSens a b -> ThSens a b
+intersectSens s1 s2 = let
+  sens1 = map sentence $ OMap.elems s1
+  sens2 = map sentence $ OMap.elems s2
+  sensI = Set.intersection (Set.fromList sens1) (Set.fromList sens2)
+ in toThSens $ map (AS_Anno.makeNamed "") $ Set.toList sensI
```

Comment:
For each sentence appearing in the intersection, we have `n` (possibly different) names of the sentence in the `n` OMS to be intersected. We could append these `n` names to a new name, and take a "union" insofar as duplications in the list of `n` names are removed first.